src/rviz/InteractiveMarkerViewer.cpp

```diff
@@ -168,34 +164,52 @@ void InteractiveMarkerViewer::update()
 {
   std::lock_guard<std::mutex> lock(mMutex);
 
-  for (auto it = std::begin(mSkeletonMarkers);
-       it != std::end(mSkeletonMarkers);)
+  if (mWorld)
   {
-    const dart::dynamics::SkeletonPtr skeleton = (*it)->getSkeleton();
-
-    if (skeleton)
+    // Update SkeletonMarkers from the world.
+    for (std::size_t i = 0; i < mWorld->getNumSkeletons(); ++i)
     {
-      std::unique_lock<std::mutex> skeleton_lock(
-          skeleton->getMutex(), std::try_to_lock);
-      if (skeleton_lock.owns_lock())
-        (*it)->update();
-
-      ++it;
+      const dart::dynamics::SkeletonPtr skeleton = mWorld->getSkeleton(i);
+
+      if (skeleton)
+      {
+        // Either a new SkeletonMarker or a previously-inserted SkeletonMarker
+        auto result = mSkeletonMarkers.emplace(
+            skeleton,
+            std::make_shared<SkeletonMarker>(
+                nullptr, &mMarkerServer, skeleton, mFrameId));
+
+        std::unique_lock<std::mutex> skeleton_lock(
+            skeleton->getMutex(), std::try_to_lock);
+        if (skeleton_lock.owns_lock())
+          result.first->second->update();
+      }
     }
-    // Skeleton no longer exists. Delete the marker.
-    else
+
+    // Clear removed skeletons
+    auto it = std::begin(mSkeletonMarkers);
+    while (it != std::end(mSkeletonMarkers))
     {
-      it = mSkeletonMarkers.erase(it);
+      // Skeleton still exists in the World, do nothing.
+      if (mWorld->hasSkeleton(it->first))
+      {
+        ++it;
+      }
+      // Skeleton does not exist. Delete our existing ShapeFrameMarker.
+      else
+      {
+        it = mSkeletonMarkers.erase(it);
+      }
     }
```

Comment:
To confirm: we don't need to update `mSkeletonMarkers`even if we don't have `mWorld`?